Utility.cpp

```diff
@@ -43,66 +49,82 @@ std::string getCharacterStats(Character* ch)
+    else if( auto* dragon = dynamic_cast<Dragon*>(character) )
     {
         //dragons don't need defensive items
     }  
 }
-void useHelpfulItem(Character*, Item* item)
+void useHelpfulItem(Character* character, Item* item)
 {
-    if( auto* ch = dynamic_cast<Dwarf*>(character) )
+    if( auto* dwarf = dynamic_cast<Dwarf*>(character) )
     {
-        ch->boostHitPoints(item->getBoost() * 2);
+        dwarf->boostHitPoints(item->getBoost() * 2);
     }
-    else if( auto* ch = dynamic_cast<Paladin*>(character) )
+    else if( auto* paladin = dynamic_cast<Paladin*>(character) )
     {
-        ch->boostHitPoints(item->getBoost() * 1.5);
+        paladin->boostHitPoints(item->getBoost() * 1.5);
     }
-    else if( auto* ch = dynamic_cast<DragonSlayer*>(character))
+    else if( auto* dragonSlayer = dynamic_cast<DragonSlayer*>(character))
     {
-        ch->boostHitPoints(item->getBoost() * 1.25);
+        dragonSlayer->boostHitPoints(item->getBoost() * 1.25);
     }
-    else if( auto* ch = dynamic_cast<Dragon*>(character) )
+    else if( auto* dragon = dynamic_cast<Dragon*>(character) )
     {
         //dragons don't carry helpful items!
     }
 }
-void useAttackItem(Character*, Item* item)
+
+
+void useAttackItem(Character* character, Item* item)
 {
-    if( auto* ch = dynamic_cast<Dwarf*>(character) )
+    if( auto* dwarf = dynamic_cast<Dwarf*>(character) )
     {
-        ch->boostAttackDamage(item->getBoost() * 1.5);
+        dwarf->boostAttackDamage(item->getBoost() * 1.5);
     }
-    else if( auto* ch = dynamic_cast<Paladin*>(character) )
+    else if( auto* paladin = dynamic_cast<Paladin*>(character) )
     {
-        ch->boostAttackDamage(item->getBoost() * 1.33);
+        paladin->boostAttackDamage(item->getBoost() * 1.33);
     }
-    else if( auto* ch = dynamic_cast<DragonSlayer*>(character))
+    else if( auto* dragonSlayer = dynamic_cast<DragonSlayer*>(character))
     {
-        assert(false);
         //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
         //so their attack item should boost their attack damage by a factor of 10
         //this means you need to GET the attack damage, multiply it by the item's boost, and BOOST the attackDamage with that multiplied value.  
         //check Character.h for available member functions you can use.
+        int attackDamage = dragonSlayer->getAttackDamage();
+        dragonSlayer->boostAttackDamage(attackDamage * item->getBoost());
     }
-    else if( auto* ch = dynamic_cast<Dragon*>(character) )
+    else if( auto* dragon = dynamic_cast<Dragon*>(character) )
     {
         //dragons don't carry attack items!
     }
-}
\ No newline at end of file
+}
+
+static std::default_random_engine randomEngine {static_cast<unsigned>(std::chrono::system_clock::now().time_since_epoch().count())};
+
+static std::uniform_int_distribution<int> itemDistribution {1, 7};
+
+int chooseItemCount()
+{
+    return itemDistribution(randomEngine);
+}
```

Comment:
Casting has not been covered (it's covered in the next project).
Statics have not been covered
- [ ] remove this.